src/Shared/Internal/RendererReEntrantManager.cs

```diff
@@ -0,0 +1,120 @@
+using System;
+#if NET35
+using System.Web;
+#else
+using System.Threading;
+#endif
+
+namespace NLog.Web.Internal
+{
+    /// <summary>
+    /// Manages if a LayoutRenderer can be called recursively.
+    /// Especially used by AspNetSessionValueLayoutRenderer
+    /// 
+    /// Since NET 35 does not support AsyncLocal or even ThreadLocal
+    /// a different technique must be used for that platform
+    /// </summary>
+    internal class RendererReEntrantManager : IDisposable
```

Comment:
Can you change `internal class` to `internal struct` ? And remove `protected virtual void Dispose(bool disposing)`, so `void Dispose()` calls `DisposalImpl()` ?

Reply:
ok

Reply:
changes uploaded